### ISSUE TITLE
Dependency error during mobile local development

### ISSUE THREAD

**Issue by rajdip-b** [TryQuiet/quiet#2113]
The `REAMME.md` in `packages/mobile/` specifies a set of commands to run in order to be able to run the mobile app locally for development However, the process fails due to some dependency issue:
![image](https://github.com/TryQuiet/quiet/assets/83924254/b9d13ce2-0ad9-4c0a-b394-f137bf74c739)

Steps to reproduce:
- Try running the mobile application as mentioned in the README.md

**Comment by holmesworcester** [comment]
@siepra any ideas?

**Comment by siepra** [comment]
As the logs states: maven throws HTTP 403 when trying to download the dependency from tor-android-binary.
Merge this branch into yours and it should work https://github.com/TryQuiet/quiet/pull/2024
I think we'll merge in into develop shortly

**Comment by rajdip-b** [comment]
@siepra Sure.

**Comment by rajdip-b** [comment]
Hey @siepra, I tried merging the latest `develop` branch and I got yet another error:
![image](https://github.com/TryQuiet/quiet/assets/83924254/d0c8a905-2172-4fa7-a249-42658aeee618)

Any ideas?

**Comment by holmesworcester** [comment]
@rajdip-b siepra will be back online on Monday CEST. Thanks for pushing on this!

**Comment by siepra** [comment]
@rajdip-b You got "failed: connect timed out" when trying to fetch the dependency from Maven repo. Make sure your internet connection is solid and Maven is not down. Maybe wait a while and then try again.
When it comes to the android-specific log output, the very error message appears somewhere around the middle of the console window, just above the:
```
> Try:
> Run with --stacktrace option to get the stack trace.
```

@holmesworcester I replaced the "bug" label with "support". It seems way more accurate.

**Comment by siepra** [comment]
@rajdip-b I've just checked by cleaning my gradle cache and building the project and it worked fine with the dependencies from `develop` branch. If the problem remains you may try downloading some other version of `tor-android`. Although I guess it might be a problem with your access to the Maven central, as you also didn't succeed with downloading `tor-android-binary` as well.
Try pausing any VPN you use, drop Tor, check your firewall etc.

**Comment by holmesworcester** [comment]
@siepra our goal is that the instructions in the README always work deterministically and we should take full responsibility for that once we know some step in the process is flakey. 

@rajdip-b have you ruled out any problems on your end?

What are our options here if this repo is flakey?

1. We could contact our friends at Guardian Project and work with them to track down the problem.
2. We could mirror the files in a location we control. For example, we can add them to the repo itself or our S3 bucket, and update them as part of the build process in a way such that builds will still work if the repo is not available.
3. Any other options?

**Comment by rajdip-b** [comment]
I tried again, seems like the issue is with the dependency URL since my internet works fine. One more thing I would like to point out is, it takes nearly 15 minutes to get the environment up and running.

**Comment by holmesworcester** [comment]
We discussed this yesterday and @kowalski reported that they have run into similar problems on another project and that Maven Central is just flaky sometimes for some users and not others. For example, it's possible that they're rate limiting your external IP address or IP range. I'm creating an issue for the Guardian Project folks to see what they think.

Stuff that might be worth trying:

1. **Try a mirror**. There are mirrors of Maven Central that might be more reliable. See: [Guide to configuring mirrors](https://maven.apache.org/guides/mini/guide-mirror-settings.html)  & [List of known mirrors](https://repo.maven.apache.org/maven2/.meta/repository-metadata.xml).
2. **Keep trying periodically**. I saw posts about Maven Central being overloaded and they just had [search downtime](https://status.maven.org/) so the problem might resolve itself.
3. **Try a VPN**. This is a janky solution, but if we're able to use it without errors and you're running into errors, maybe a VPN would help?

**Comment by siepra** [comment]
`android-jsc-intl` (a javascript core) is bound with react-native, not guardian project so I'm confused with this link you tried, especially that guardian projects' maven repo indeed doesn't host anything under this path (https://github.com/guardianproject/gpmaven/tree/master).

My hypothesis is environment setup hasn't been done right and now gradle confuses maven sources while resolving dependencies.

@rajdip-b It'd be wonderful if you listed the exact steps you take to run the project.

**Comment by rajdip-b** [comment]
> > it takes nearly 15 minutes to get the environment up and running
> 
> Setting up the environment time depends on the person who performs it (installing SDKs, setting envs, etc.). Are you sure you didn't mean the initial build time? It will get better with the following builds as there'll be gradle deamons and cache available so don't worry. It also depends on your hardware profile of course.
> 
> Anyway, I'm glad it didn't discourage you!

Yeah sorry, my bad. I meant the initial build. But the hardware might not ne the issue, since mostly it was downloading a lot of stuff.

**Comment by EmiM** [comment]
@rajdip-b does the problem still happen to you?
